pkg/resource/deploy/step.go

```diff
@@ -1439,7 +1439,7 @@ func (s *ImportStep) Apply() (resource.Status, StepCompleteFunc, error) {
 	s.diffs, s.detailedDiff = diff.ChangedKeys, diff.DetailedDiff
 
 	if diff.Changes != plugin.DiffNone {
-		const message = "inputs to import do not match the existing resource"
+		message := fmt.Sprintf("inputs to import do not match the existing resource: %v", s.diffs)
```

Comment:
Should we pretty print these?  Maybe something like (completely untested):
```suggestion
		message := fmt.Sprintf("inputs to import for %s do not match the existing resource for the following properties:", s.new.URN)
		for _, diff := range s.diffs {
			message = fmt.Sprintf("%s\n\t- %v", message, diff)
		}
```

So both the urn of the resource that is failing is included, and the properties are printed a bit nicer.  Maybe we could even pretty print the detailed diff so the user knows what they would have to change?

Reply:
I'll try that out!

Reply:
Would love to see a screenshot when you do also! Also, I'm not sure what effect this has, but we should test with `--non-interactive` and `--diff` to see if these changes work there too (again, not an expert on diagnostics yet so this might be no big deal/have no impact whatsoever, but always good to check!)

Reply:
![image](https://github.com/user-attachments/assets/ff128617-65ce-4c4c-a448-249e9fa7044d)
I like the formatting, I also sort of like adding in the detailed diff translated into strings - but I wonder if it might not be confusing (for example, with this role I'm testing with, when I alter the assumeRolePolicy, it shows up as updated, but the managed_policy_arns shows as added, but they are both there in the resource I'm trying to import) - unless theres a better way to translate DiffKind?

Reply:
![image](https://github.com/user-attachments/assets/0194c447-ff27-4f5b-995a-cda402033979)
alternative with no DetailedDiff

Reply:
The diagnostic stream is already tied to a URN (see first arg to diag.StreamMessage), do we really it in the message as well?

Reply:
good point! I hadn't noticed it there actually but I see it now in blue `aws:iam:Role (imported_role)`. 
Well, on the formatting front, and to Will's point - looks like `--non-interactive` does not play nice with multiline messages, haven't tracked done exactly why yet... 
![image](https://github.com/user-attachments/assets/f61e7fc1-eadb-4a3d-b03f-6073c45b159f)
but both could be arguments for the current change just putting the input keys into the existing message `message := fmt.Sprintf("inputs to import do not match the existing resource: %v", s.diffs)`

Reply:
this is how it looks with the current code 
`pulumi preview`
![image](https://github.com/user-attachments/assets/4338df17-6f70-4ab0-9d57-6c3fdb951bc2)
`pulumi preview --non-interactive`
![image](https://github.com/user-attachments/assets/c4464d41-617a-4064-bf08-1b74ae027e7a)
`pulumi preview --diff`
![image](https://github.com/user-attachments/assets/4f36cf8d-c432-4ba7-a9fc-64b12d3eb5fe)
`pulumi up`
![image](https://github.com/user-attachments/assets/731cee9f-1d96-4282-89d2-a144dff03990)

Reply:
I'm open to whatever y'all think re: formatting and whether to include more details besides just the keys. Also - I notice the printed out keys `[assumeRolePolicy managedPolicyArns]` don't match the python format `assume_role_policy` - wonder if that would be confusing or not to users